apps/system/js/browser_context_menu.js

```diff
@@ -51,6 +50,8 @@
         return;
       }
 
+      SettingsListener.observe(PINNING_PREF, '', this._setPinning.bind(this));
+
```

Comment:
The problem here will be the same than the original bug. The browser_context_menu module get's instantiated many times, even before shown, so we can have several listeners to settings running on the same preference.